From 268bfad97a06ffb665b8ebda2a4ba870ac6541e8 Mon Sep 17 00:00:00 2001
Message-Id: <268bfad97a06ffb665b8ebda2a4ba870ac6541e8.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Pallavi <pallavi.g@intel.com>
Date: Thu, 30 May 2013 13:28:56 +0530
Subject: [PATCH 033/312] FOR_UPSTREAM [VPG]: drm/i915: Generalized the Panel
 Fitter enabling

Modified the Panel fitter enabling logic to be generic across the pipes

Issue: APDEV-1164
OTC-Tracker: VIZ-3480
Change-Id: I066bb038c5029c1002dec9ca7835b9db77c2fe5f
Signed-off-by: Pallavi <pallavi.g@intel.com>
Reviewed-on: http://android.intel.com:8080/110791
Reviewed-by: Wood, Brian J <brian.j.wood@intel.com>
Reviewed-by: Stimson, Dale B <dale.b.stimson@intel.com>
Tested-by: Stimson, Dale B <dale.b.stimson@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
(cherry picked from commit ab067e3f4e8f46967c3c67b72f4a6e9344c0450d)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h   |    7 ++++---
 drivers/gpu/drm/i915/intel_hdmi.c |   31 +++++++++++++++++++------------
 2 files changed, 23 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 3f878f7..9ad8d72 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -562,9 +562,10 @@ struct intel_device_info {
 #undef SEP_SEMICOLON
 
 enum hdmi_panel_fitter {
-	AUTO_SCALE,
-	PILLAR_BOX,
-	LETTER_BOX,
+	PFIT_OFF,
+	AUTOSCALE,
+	PILLARBOX,
+	LETTERBOX,
 };
 
 enum i915_cache_level {
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index cea534a..891b621 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -693,12 +693,15 @@ static void intel_hdmi_prepare(struct intel_encoder *encoder)
 
 	if (intel_hdmi->pfit) {
 		u32 val = 0;
-		if (intel_hdmi->pfit == PILLAR_BOX)
-			val =  PFIT_ENABLE | (1 << PFIT_PIPE_SHIFT) |
-				PFIT_SCALING_PILLAR;
-		else if (intel_hdmi->pfit == LETTER_BOX)
-			val =  PFIT_ENABLE | (1 << PFIT_PIPE_SHIFT) |
-				PFIT_SCALING_LETTER;
+		if (intel_hdmi->pfit == AUTOSCALE)
+			val =  PFIT_ENABLE | (crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_AUTO;
+		if (intel_hdmi->pfit == PILLARBOX)
+			val =  PFIT_ENABLE | (crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_PILLAR;
+		else if (intel_hdmi->pfit == LETTERBOX)
+			val =  PFIT_ENABLE | (crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_LETTER;
 		DRM_DEBUG_DRIVER("pfit val = %x", val);
 
 		I915_WRITE(PFIT_CONTROL, val);
@@ -781,6 +784,7 @@ static void intel_enable_hdmi(struct intel_encoder *encoder)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct intel_crtc *intel_crtc = to_intel_crtc(encoder->base.crtc);
 	struct intel_hdmi *intel_hdmi = enc_to_intel_hdmi(&encoder->base);
+	struct intel_crtc *crtc = to_intel_crtc(encoder->base.crtc);
 	u32 temp;
 	u32 enable_bits = SDVO_ENABLE;
 
@@ -809,12 +813,15 @@ static void intel_enable_hdmi(struct intel_encoder *encoder)
 
 	if (intel_hdmi->pfit) {
 		u32 val = 0;
-		if (intel_hdmi->pfit == PILLAR_BOX)
-			val =  PFIT_ENABLE | (1 << PFIT_PIPE_SHIFT) |
-				PFIT_SCALING_PILLAR;
-		else if (intel_hdmi->pfit == LETTER_BOX)
-			val =  PFIT_ENABLE | (1 << PFIT_PIPE_SHIFT) |
-				PFIT_SCALING_LETTER;
+		if (intel_hdmi->pfit == AUTOSCALE)
+			val =  PFIT_ENABLE | (intel_crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_AUTO;
+		if (intel_hdmi->pfit == PILLARBOX)
+			val =  PFIT_ENABLE | (intel_crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_PILLAR;
+		else if (intel_hdmi->pfit == LETTERBOX)
+			val =  PFIT_ENABLE | (intel_crtc->pipe <<
+				PFIT_PIPE_SHIFT) | PFIT_SCALING_LETTER;
 		DRM_DEBUG_DRIVER("pfit val = %x", val);
 
 		I915_WRITE(PFIT_CONTROL, val);
-- 
1.7.9.5

